library/src/main/java/com/okta/oidc/clients/web/WebAuthClientImpl.java

```diff
@@ -130,10 +132,23 @@ public boolean isInProgress() {
     public void signIn(@NonNull final Activity activity, AuthenticationPayload payload) {
         registerActivityLifeCycle(activity);
         cancelFuture();
+        //add the login hint if it exists.
+        if (payload == null && mLoginHint != null) {
```

Comment:
`TextUtils.isEmpty` instead of null checking? Or is empty string a valid login hint?

Reply:
Empty string is valid. This is to pre-fill the username in the SIW for the web view. so it can be empty.